### PR TITLE
clarify Trainer running state atribs.

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

moving attributes to the main class as they won't be deprecated

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by tchaton — APPROVED** [review]
LGTM ! Can you add it for tuning, evaluating ?

**Comment by Borda** [comment]
> LGTM ! Can you add it for tuning, evaluating ?

done, in fact as the new API stays, I moved it to Trainer

**Comment by Borda** [comment]
Well not sure why these three tests are failing, any suggestion? @awaelchli 
```
FAILED tests/checkpointing/test_model_checkpoint.py::test_checkpoint_repeated_strategy_extended[False]
FAILED tests/checkpointing/test_model_checkpoint.py::test_checkpoint_repeated_strategy_extended[True]
FAILED tests/loggers/test_csv.py::test_file_logger_log_hyperparams - yaml.con...
```

**Comment by carmocca** [comment]
I don't like the `set_*` functions. Users might think this changes the Trainer state (as in, to switch from training to evaluation) when it only changes a property.

**Comment by Borda** [comment]
> I don't like the `set_*` functions. Users might think this changes the Trainer state (as in, to switch from training to evaluation) when it only changes a property.

Ok, so what else do you suggest?

**Comment by carmocca** [comment]
```python
@is_testing.setter
def is_testing(self, value: bool):
    if value:
        self._running_stage = RunningStage.TESTING
```

**Comment by Borda** [comment]
> ```python
> @is_testing.setter
> def is_testing(self, value: bool):
>     if value:
>         self._running_stage = RunningStage.TESTING
> ```

Yes, that is standard setter, but with this one is unclear what it does if you pass something else then true/1
We want something which does only one thing...

**Review by tchaton — APPROVED** [review]
LGTM !

**Comment by Borda** [comment]
ok, so @awaelchli suggests keeping it as attributes, correct?
my motivation was to clearly distinguish that it is a state attribute, but you have good point with mapping PT attributes

**Comment by awaelchli** [comment]
I guess my main concern was just that you are introducing two names, is_testing and set_testing, while in python we do not often see this kind of getter and setter naming. the convention is to have the same name for the setter and getter (typically). Please correct me if I'm wrong. It's not a strong opinion, but I see some benefits having closer naming to pytorch :)

**Comment by Borda** [comment]
@awaelchli I gree that introducing new attributes is not a good way :]
 one point/question you have referred that `nn.Module` has training/testing but won't be rather confusing to have the same also for the trainer?

**Comment by awaelchli** [comment]
so far I have not experienced or seen complaints about confusion between model and trainer attributes. But this is mainly because nn.Module only has a .training attribute (no testing) and on the other hand our Trainer has only a .testing attribute and no training. So, it could be seen as confusing from different perspectives :laughing:

**Comment by Borda** [comment]
> so far I have not experienced or seen complaints about confusion between model and trainer attributes. But this is mainly because nn.Module only has a .training attribute (no testing) and on the other hand our Trainer has only a .testing attribute and no training. So, it could be seen as confusing from different perspectives 😆

ok, I have reverted the rename and just moved the properties to the trainer as they won't be deprecated

**Review by carmocca — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by rohitgr7 — APPROVED** [review]
